syntax = "proto3";

option go_package = "github.com/webitel/voicebot;gen";

package voicebot;

// The VoiceBot service definition
service VoiceBot {
  // Bidirectional streaming RPC for conversation
  rpc Converse (stream AudioRequest) returns (stream AudioResponse) {}
}

// The request message containing audio data
message AudioRequest {
  string conversation_id = 1;
  bytes audio_data = 2;
}

// The response message containing audio data
message AudioResponse {
  string conversation_id = 1;
  bytes audio_data = 2;
  bool stop_talk = 3;
}